src/cls/IPM/General/LockFile.cls

```diff
@@ -0,0 +1,117 @@
+
+ClassMethod CreateLockFileForModule(
+	baseModName As %String,
+	ByRef params)
+{
+    set verbose = $get(params("Verbose"), 0)
+    set module = ##class(%IPM.Storage.Module).NameOpen(baseModName, .sc)
+    $$$ThrowOnError(sc)
+
+    if (verbose) {
+        write !, "Creating lock file for module "_module.DisplayName
+    }
+
+    /// Create lock file object and set values for the base module
+    set lockFile = ##class(%IPM.General.LockFile).%New()
+    set lockFile.ModuleName = module.Name
+    set lockFile.VersionString = module.VersionString
+    set lockFile.Repository = module.Repository
+
+    // Create dependency list of least dependent module to most dependent from dependency graph
+    $$$ThrowOnError(module.BuildDependencyGraph(.dependencyGraph))
+    do ##class(%IPM.Utils.Module).ConstructInvertedDependencyGraph(.dependencyGraph, .inverted)
+    set dependencyList = ##class(%IPM.Utils.Module).GetFlatDependencyListFromInvertedDependencyGraph(.inverted)
+
+    // Iterate over module dependency list
+    // For each module:
+    //      1. Set values for module name, version, repository (name), and direct dependencies
+    //      2. Add repository information to the list (if the module is from a new repo)
+    set moduleName = ""
+    for i = 1:1:dependencyList.Count() {
+        set moduleName = dependencyList.GetAt(i).Name
+        set mod = ##class(%IPM.Storage.Module).NameOpen(moduleName, .sc)
+        $$$ThrowOnError(sc)
+
+        if (verbose) {
+            write !, "Adding "_mod.DisplayName_" to the lock file"
+        }
+
+        // Add the dependency to the lock file
+        set dependencyVal = ##class(%IPM.General.LockFile.Dependency).%New()
+        set dependencyVal.VersionString = mod.VersionString
+        set dependencyVal.Repository = mod.Repository
+        set depKey = ""
+        for {
+            set depMod = mod.Dependencies.GetNext(.depKey)
+            quit:depKey=""
+            $$$ThrowOnError(dependencyVal.Dependencies.SetAt(depMod.VersionString, depMod.Name))
+        }
+        $$$ThrowOnError(lockFile.Dependencies.SetAt(dependencyVal, mod.Name))
+
+        // Add the dependency's repository to the lock file
+        do AddRepositoryToLockFile(.lockFile, mod.Repository, verbose)
+    }
+    // Add repository for base module if not already added by a dependency
+    // Skip undefined repositories as that means the module was installed via the zpm "load" command
+    if (module.Repository '= "") {
+        do AddRepositoryToLockFile(.lockFile, module.Repository, verbose)
+    }
+
+    $$$ThrowOnError(lockFile.%JSONExportToStream(.lockFileJSON, "LockFileMapping"))
+
+    set lockFilePath = module.Root_..#LockFileName
+    if (verbose) {
+        write !, "Saving lock file for "_module.Name_" to: "_lockFilePath
+    }
+    set file = ##class(%File).%New(lockFilePath)
+    $$$ThrowOnError(file.Open("WSN"))
+    $$$ThrowOnError(file.CopyFrom(lockFileJSON))
```

Comment:
Need to save file I believe. May want to use %Stream.FileCharacter instead of using file instead